CARDS-2586: YE new system quality and infection control questions to be added to existing PREMs surveys

sashaandjic opened the pull request:
Added system quality and infection control questions to `OAIP`, `OED `and `Rehab `xml files.

sashaandjic committed:
CARDS-2586: YE new system quality, infection control, demographic questions to be added to existing PREMs surveys
OAIP.hml - added system quality and infection control questions

sashaandjic committed:
CARDS-2586: YE new system quality, infection control, demographic questions to be added to existing PREMs surveys
OED.xml: added system quality and infection control questions

sashaandjic committed:
CARDS-2586: YE new system quality, infection control, demographic questions to be added to existing PREMs surveys
Fixed spelling errors in OED.xml

sashaandjic committed:
CARDS-2586: YE new system quality, infection control, demographic questions to be added to existing PREMs surveys
EOD.xml: added missing minAnswers=1

sashaandjic committed:
CARDS-2586: YE new system quality, infection control, demographic questions to be added to existing PREMs surveys
Rehab.hml: Added system quality and infection control questions

sashaandjic committed:
CARDS-2586: YE new system quality, infection control, demographic questions to be added to existing PREMs surveys
OAIP.xml and OED.xml: removed dot from the infection control practices section title

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/OAIP.xml:
```diff
@@ -1802,5 +1802,796 @@
+		<primaryNodeType>cards:Section</primaryNodeType>
+		<property>
+			<name>label</name>
+			<value>The following questions ask you about UHN’s infection control practices</value>
```
To make the "infection control" section available only if the visit occurred in January through March, we need to add:

* a hidden reference question `visit_date` at the beginning of the survey, after `hospital` and `department`
* a hidden computed question `visit_month` right after `visit_date`, calculated from `visit_date` as a number 0-11; something like `return new Date(@{visit_date}).getMonth()` might work (not tested) - let's hope we have no issues with backend js limitations
* a condition to this section, to be displayed only if `visit_month` is between 0 and 2.

The above should be done for all 3 surveys containing infection control questions.

marta- commented:
> TO DO:
> 
> * add conditional to the infection control section to show up only if visit is in Jan, Feb or March
> * demographic

Demographic will likely be a separate survey, out of scope for this iteration.

How to handle the conditional is outlined in the inline comment above.

sashaandjic committed:
YE new system quality, infection control, demographic questions to be added to existing PREMs surveys
For OAIP and OED: Added condition to the infection control section to appear for visit months Jan - Mar

sashaandjic committed:
CARDS-2586: YE new system quality, infection control, demographic questions to be added to existing PREMs surveys
Rehab.xml: Added condition to the infection control section to appear for visit months Jan - Mar

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Rehab.xml:
```diff
@@ -2044,32 +2097,54 @@
+		</node>
+	</node>
+	<node>
+		<name>section_rs_sqp_3</name>
```
This section name is inconsistent with the rest of the form. Let's change it to `Section_sqp_3`.

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Rehab.xml:
```diff
@@ -2262,41 +2337,918 @@
+		</node>
+	</node>
+	<node>
+		<name>section_rs_sqp_5</name>
```
This section name is inconsistent with the rest of the form. Let's change it to Section_sqp_5.
```suggestion
		<name>Section_sqp_5</name>
```

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Rehab.xml:
```diff
@@ -2262,41 +2337,918 @@
+		</node>
+	</node>
+	<node>
+		<name>rs_module4</name>
```
Can we just name this Section_ic_1?

sashaandjic replied:
Section ic_1 is right below this one. This is unfortunately the way it was implemented before and I followed the pattern. Each module starts with module#, but then each subsequent section is named after the question. I would like to modify all of the section names, but I am afraid if this will break something in their pipeline. All other suggestions, except this one, were implemented.

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Rehab.xml:
```diff
@@ -2262,41 +2337,918 @@
+		</node>
+	</node>
+	<node>
+		<name>Section_ic_1</name>
```
... and this Section_ic_2? Otherwise the section name for question is_2 is very confusing.
```suggestion
		<name>Section_ic_2</name>
```

marta- reviewed (changes requested):
Looks good to me, some section name tweaks are suggested below.

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/OED.xml:
```diff
@@ -99,6 +99,64 @@
 			</property>
 		</node>
+		<node>
+			<name>oed_visit_date</name>
```
Add question text: Visit Date

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/OED.xml:
```diff
@@ -99,6 +99,64 @@
+			</property>
+		</node>
+		<node>
+			<name>oed_visit_month</name>
```
Add question text: Visit month

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Rehab.xml:
```diff
@@ -128,6 +128,64 @@
 			</property>
 		</node>
+		<node>
+			<name>rs_visit_date</name>
```
Add question text: Visit date

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Rehab.xml:
```diff
@@ -128,6 +128,64 @@
+			</property>
+		</node>
+		<node>
+			<name>rs_visit_month</name>
```
Add question text: Visit month

sashaandjic committed:
Update prems-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Rehab.xml

Co-authored-by: Marta Girdea <651980+marta-@users.noreply.github.com>

sashaandjic committed:
Update prems-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Rehab.xml

Co-authored-by: Marta Girdea <651980+marta-@users.noreply.github.com>

sashaandjic committed:
CARDS-2586: YE new system quality, infection control, demographic questions to be added to existing PREMs surveys
Added Visit date and Visit month labels as per review request.

Hajany reviewed (approved):
tested, looks good

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Rehab.xml:
```diff
@@ -1843,15 +1911,10 @@
 	</node>
 	<node>
-		<name>rs_module3</name>
+		<name>Section_sqp_1</name>
```
The naming of these new sections is so inconsistent...

sdumitriu reviewed (approved):
There is a patient-visible problem with the missing Month hidden answer, the ED/IP forms first show as having 2 pages, then jump to 3 after saving the first page.
